primitive[] primitives
string camera_frame
